renting_platform.sql

```diff
@@ -0,0 +1,159 @@
+-- Created by Vertabelo (http://vertabelo.com)
+-- Last modification date: 2022-09-12 14:31:40.494
+
+-- tables
+-- Table: adress
+CREATE TABLE adress (
+    id int NOT NULL AUTO_INCREMENT,
+    street varchar(45) NOT NULL,
+    house_number int NULL,
+    city_id int NOT NULL,
+    CONSTRAINT adress_pk PRIMARY KEY (id)
+);
+
+-- Table: cash_withdrawal_from_lessee_to_service
+CREATE TABLE cash_withdrawal_from_lessee_to_service (
+    id int NOT NULL,
+    price int NOT NULL,
+    reservation_id int NOT NULL,
+    CONSTRAINT cash_withdrawal_from_lessee_to_service_pk PRIMARY KEY (id)
+);
+
+-- Table: cash_withdrawal_from_service_to_owner
+CREATE TABLE cash_withdrawal_from_service_to_owner (
+    id int NOT NULL,
+    price int NOT NULL,
+    cash_withdrawal_from_lessee_to_service_id int NOT NULL,
+    CONSTRAINT cash_withdrawal_from_service_to_owner_pk PRIMARY KEY (id)
+);
+
+-- Table: city
+CREATE TABLE city (
+    id int NOT NULL AUTO_INCREMENT,
+    name varchar(35) NOT NULL,
+    region_id int NOT NULL,
+    CONSTRAINT city_pk PRIMARY KEY (id)
+);
+
+-- Table: dwelling
+CREATE TABLE dwelling (
+    id int NOT NULL AUTO_INCREMENT,
+    area int NOT NULL,
+    floor int NOT NULL,
+    rooms_number int NOT NULL,
+    adress_id int NOT NULL,
+    dwelling_owner_id int NOT NULL,
+    reservation_id int NOT NULL,
+    description varchar(100) NOT NULL,
+    CONSTRAINT dwelling_pk PRIMARY KEY (id)
+);
+
+-- Table: dwelling_owner
+CREATE TABLE dwelling_owner (
+    id int NOT NULL AUTO_INCREMENT,
+    name varchar(25) NOT NULL,
+    surname varchar(50) NOT NULL,
+    phome varchar(12) NOT NULL,
+    email varchar(35) NOT NULL,
```

Comment:
Also, maybe all these phones and emails in your user types should be unique? But how do you make them unique across all of your tables related to user accounts? Maybe you should use an "abstract" `user` table that has one-to-one with other user tables like `dwelling_owner` and `lessee`?

Reply:
SO I should create new table(user) which has unique fields ( phone and email (both can be NULL) ) and which has one-to-one with dwelling_owner and lessee? And for example when I want enter dwelling_owner`s phone I must leave lessee_id=NULL which u can see on the picture?:
![image](https://user-images.githubusercontent.com/91155188/190018040-3622a646-1eb4-4984-8e00-da3bacf276cc.png)

Reply:
1. I don't think you would want a user without email (e.g. for authentication purposes), so maybe not NULL. But that's your DB :)
2. In this case, `user_id` should be stored in `dwelling_owner` and `lessee` tables. Because a dwelling_owner and a lessee are both users, but a user cannot be both dwelling_owner and lessee, right? To avoid NULLs in your FK, it's better to store a user_id in those two tables rather than storing dwelling_owner_id and lessee_id in `user`.
Try to think of tables as classes: DwellingOwner and Lessee would inherit `email` and `phone` fields from User class

Reply:
Yep, Thanks